packages/portal-proto/src/components/InputEntityList/MatchedTable.tsx

```diff
@@ -92,13 +82,28 @@ const MatchedTable = ({
               meta: {
                 sortingFn: (rowA, rowB) => {
                   const property = `submitted_${id.replaceAll(".", "_")}`;
-                  if (rowA[property] > rowB[property]) {
-                    return 1;
-                  }
-                  if (rowA[property] < rowB[property]) {
-                    return -1;
+
+                  const valueA = rowA[property];
+                  const valueB = rowB[property];
+
+                  // if values are undefined i.e., "--"
+                  if (!valueA && !valueB) return 0;
+                  if (!valueA) return 1;
+                  if (!valueB) return -1;
+
+                  const numA = Number(valueA);
+                  const numB = Number(valueB);
+
+                  const isNumA = !isNaN(numA);
+                  const isNumB = !isNaN(numB);
```

Comment:
you are way braver than i am to trust `isNan` over `Number.isNaN` 😆

Reply:
I found this:
1. `isNaN()` will convert it to the number first.
2. `Number.isNaN()` doesn't.

Since I am already using `Number() `for that we can use `Number.isNaN()`.